File: Crawler/src/entities/DomainGroup.ts
import { Entity, Column, PrimaryGeneratedColumn, OneToMany } from "typeorm";
import { Domain } from "./Domain";
import { Route, IndexingConfig } from "./Route";

@Entity("crawler.DomainGroups")
export class DomainGroup {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ update: false })
  name: string;

  @Column({ type: "simple-json", update: false })
  indexingConfig: IndexingConfig;

  @OneToMany(() => Domain, domain => domain.domainGroup)
  domains: Domain[];

  @OneToMany(() => Route, route => route.domainGroup)
  routes: Route[];

  constructor(props?: Partial<DomainGroup>) {
    Object.assign(this, props);
  }
}

File: QA.Search.Data/Models/DomainGroup.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace QA.Search.Data.Models
{
    public class DomainGroup
    {
        public int Id { get; set; }
        [MaxLength(450)]
        public string Name { get; set; }
        public string IndexingConfig { get; set; } = "null";

        public ICollection<Domain> Domains { get; set; } = Array.Empty<Domain>();
        public ICollection<Route> Routes { get; set; } = Array.Empty<Route>();
    }
}
